code/ui/blocks/src/blocks/Source.tsx

```diff
@@ -94,50 +103,52 @@ export const useSourceProps = (
   docsContext: DocsContextProps<any>,
   sourceContext: SourceContextProps
 ): PureSourceProps & SourceStateProps => {
-  const { id: primaryId, parameters } = docsContext.storyById();
-
-  const codeProps = props as CodeProps;
-  const singleProps = props as SingleSourceProps;
-  const multiProps = props as MultiSourceProps;
-
-  let { format, code: source } = codeProps; // prefer user-specified code
-
-  const targetIds = multiProps.ids || [singleProps.id || primaryId];
-  const storyIds = targetIds.map((targetId) => {
-    return targetId;
-  });
-
-  const stories = useStories(storyIds, docsContext);
-  if (!stories.every(Boolean)) {
+  const storyIds = props.ids || (props.id ? [props.id] : []);
+  const storiesFromIds = useStories(storyIds, docsContext);
+  if (!storiesFromIds.every(Boolean)) {
     return { error: SourceError.SOURCE_UNAVAILABLE, state: SourceState.NONE };
   }
 
-  if (!source) {
-    // just take the format from the first story, given how they're all concatinated together...
-    // TODO: we should consider sending an event with all the sources separately, instead of concatenating them here
-    ({ format } = getStorySource(storyIds[0], sourceContext));
-    source = storyIds
-      .map((storyId, idx) => {
-        const { code: storySource } = getStorySource(storyId, sourceContext);
-        const storyObj = stories[idx] as PreparedStory;
-        return getSnippet(storySource, storyObj);
+  // The check didn't actually change the type.
+  let stories: PreparedStory[] = storiesFromIds as PreparedStory[];
+  if (props.of) {
+    const resolved = docsContext.resolveModuleExport(props.of);
```

Comment:
This needs to be changed to `resolveOf` with the new changes in `next` I believe.